aws/src/main/scala/com/gu/contentapi/client/IAMSigner.scala

```diff
@@ -51,4 +52,15 @@ class IAMSigner(credentialsProvider: AWSCredentialsProvider, awsRegion: String)
     signer.sign(requestToSign, credentialsProvider.getCredentials)
     requestToSign.getHeaders.asScala.toMap + ("Accept-Encoding" -> "identity")
   }
+
+  /**
+    * Returns the given set of headers, updated to include AWS sig4v signed headers based on the request and the credentials
+    *
+    * @param headers  Current set of request headers
+    * @param url      Request URL, including params
+    * @return         Updated set of headers, including the authorisation headers
+    */
+  def addIAMHeaders(headers: Map[String, String], url: String): Map[String, String] = {
+    addIAMHeaders(headers, new URI(url))
```

Comment:
Should this be wrapped in a `Try`?

Reply:
possibly, but I'm now thinking the best solution is to remove this method and let the app take responsibility for the `URI` construction!